Sources/Tokenizer.swift

```diff
@@ -36,10 +36,37 @@ extension String {
       components.append(word)
     }
 
-    return components
+    return smartJoin(components)
   }
 }
 
+// joins back components around characters used in variables lists and filters
+private func smartJoin(_ components: [String]) -> [String] {
```

Comment:
Does this have performance implications?

Reply:
@yonaskolb thanks for pointing it out =) I've reimplemented it, should be faster now.